### DISCUSSION TITLE
What Is the Maximum Number of Deployments the Dokku Can Handle?

### DISCUSSION THREAD

**Discussion by MahmoudMousaHamad** [dokku/dokku#5140] [Q&A] [answered]
Hello! I was wondering if anybody knows what is the largest number of deployments that a dokku instance can support?

**Comment by josegonzalez** [comment, marked as answer]
There isn't a "maximum number". The resources used by your app and the resources available to your server (or cluster, if you are deploying via our Kubernetes or Nomad plugin) decide how many running processes you can run on a given server.

**Reply by MahmoudMousaHamad** [reply]
Thank you for the answer! That's because dokku depends on docker and you can have as many containers/processes as you want as long you have enough resources to run them, correct?

**Reply by josegonzalez** [reply]
Basically. Dokku doesn't really impose any limitations on what is running or how resources are allocated (unless you constrain those via the `resources` plugin) so its kinda up to you to set those sorts of limits.

**Reply by Lewiscowles1986** [reply]
Not imposing resource limitations, is not the same thing as not having limits. While I'm sure good intentions were present when this answer was given, there will be practical limits to Dokku as with any other system.

**Reply by josegonzalez** [reply]
The limits for your app are whatever resources are available on the host if you don't set any limits via the resource plugin. You don't have limitless resources on your app. Does that make my answer more clear?

**Reply by Lewiscowles1986** [reply]
I was thinking more https://kubernetes.io/docs/setup/best-practices/cluster-large/ which perhaps was not OP intention (they marked as the answer after all); but would be what the wording of the question suggests.

So rather than artificial per-node limits, I read this as "When should I consider not using Dokku?"

**Reply by josegonzalez** [reply]
Thats a very different question than what the user asked. And if you're looking for a cluster-specific question, I think you found your answer? Feel free to ask a different question (or a more specific one) if the above answer doesn't suffice for what it is you are looking for.

**Comment by MahmoudMousaHamad** [comment]
I see thank you! That's because dokku depends on docker and you can have as
many containers as you want as long you have enough resources to run them,
correct?

**Reply by josegonzalez** [reply]
Yeah thats correct.

**Comment by MahmoudMousaHamad** [comment]
Yes, thank you :)
